src/Agent.Worker/Build/BuildJobExtension.cs

```diff
@@ -200,19 +201,22 @@ public override void InitializeJobExtension(IExecutionContext executionContext,
             executionContext.SetVariable(Constants.Variables.Pipeline.Workspace, Path.Combine(_workDirectory, trackingConfig.BuildDirectory), isFilePath: true);
         }
 
-        private bool TrySetPrimaryRepositoryAndProviderInfo(IExecutionContext executionContext)
+        private bool TryGetRepositoryInfo(IExecutionContext executionContext, string repoAlias, out RepositoryInfo repoInfo)
         {
-            // Return the first repository resource and its source provider.
+            // Return the matching repository resource and its source provider.
             Trace.Entering();
+            repoInfo = new RepositoryInfo();
             var extensionManager = HostContext.GetService<IExtensionManager>();
             List<ISourceProvider> sourceProviders = extensionManager.GetExtensions<ISourceProvider>();
-            Repository = executionContext.Repositories.SingleOrDefault();
-            if (Repository != null)
+            var repository = executionContext.Repositories.FirstOrDefault(r => string.Equals(r.Alias, repoAlias, StringComparison.OrdinalIgnoreCase));
+            if (repository != null)
             {
-                SourceProvider = sourceProviders.FirstOrDefault(x => string.Equals(x.RepositoryType, Repository.Type, StringComparison.OrdinalIgnoreCase));
+                var sourceProvider = sourceProviders.FirstOrDefault(x => string.Equals(x.RepositoryType, repository.Type, StringComparison.OrdinalIgnoreCase));
 
-                if (SourceProvider != null)
+                if (sourceProvider != null)
                 {
+                    repoInfo.Repository = repository;
+                    repoInfo.SourceProvider = sourceProvider;
                     return true;
                 }
             }
```

Comment:
It wont let me comment a line lower but do we want to set repoInfo to null in the return false case?

Reply:
I decided it made the code too messy to return null here. Better to always return the info object. We already have code to check the repo and source provider for null.